Python/chapter01/1.4 - PalinPerm/miguel_1.4_sol.py

```diff
@@ -0,0 +1,114 @@
+"""
+Python version 3.7.0
+1.4 - Palindrome Permutation
+Given a string, write a function to check if it is a permutation of a palindrome.
+A palindrome is a word or phrase that is the same forwards and backwards.  A permutation is a
+rearrangement of letters.  The palindrome does not need to be limited to just dictionary words.
+"""
+import unittest
+import itertools as it
+import collections
+from typing import Callable
+
+
+def is_permutation_of_palindrome(word: str) -> bool:
+    """
+    Checks if a word is a palindrome by analyzing the frequencies of characters.
+    A palindrome is a word or phrase that is the same forwards and backwards.
+    When determining character frequencies, word will be put to lowercase
+    and spaces will not be counted
+    A palindrome also has the following property:
+    * all characters have an even count or all except one are even
```

Comment:
a more succint/clearer way of expressing this is "at most one character appears an odd number of times".